oauth2_provider/views/mixins.py

```diff
@@ -282,11 +289,38 @@ def dispatch(self, request, *args, **kwargs):
         if not valid:
             # Alternatively allow access tokens
             # check if the request is valid and the protected resource may be accessed
-            valid, r = self.verify_request(request)
-            if valid:
-                request.resource_owner = r.user
-                return super().dispatch(request, *args, **kwargs)
-            else:
-                return HttpResponseForbidden()
+            try:
+                valid, r = self.verify_request(request)
+                if valid:
+                    request.resource_owner = r.user
+                    return super().dispatch(request, *args, **kwargs)
+            except ValueError:
+                pass
```

Comment:
This code is from the original PR; it looks wrong to me as it will squash and hide `ValueError` raised by the view we are mixing in to, which should be the responsibility of the view itself, if it wants to hide those errors.

Are we trying to squash a potential `ValueError` from `self.verify_request()`? This would make more sense, but then the `try / except` should be just over that line.